apps/meteor/ee/client/omnichannel/hooks/useOmnichannelPrioritiesMenu.tsx

```diff
@@ -0,0 +1,63 @@
+import { LivechatPriorityWeight } from '@rocket.chat/core-typings';
+import type { Menu } from '@rocket.chat/fuselage';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint, useTranslation } from '@rocket.chat/ui-contexts';
+import type { ComponentProps } from 'react';
+import React, { useMemo } from 'react';
+
+import { dispatchToastMessage } from '../../../../client/lib/toast';
+import { PriorityIcon } from '../priorities/PriorityIcon';
+import { useOmnichannelPriorities } from './useOmnichannelPriorities';
+
+export const useOmnichannelPrioritiesMenu = (rid: string): ComponentProps<typeof Menu>['options'] | Record<string, never> => {
+	const t = useTranslation();
+	const updateRoomPriority = useEndpoint('POST', '/v1/livechat/room/:rid/priority', { rid });
+	const removeRoomPriority = useEndpoint('DELETE', '/v1/livechat/room/:rid/priority', { rid });
+	const { data: priorities } = useOmnichannelPriorities();
+
+	const handlePriorityChange = useMutableCallback((priorityId: string) => async () => {
+		try {
+			priorityId ? await updateRoomPriority({ priorityId }) : await removeRoomPriority();
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	});
+
+	const renderOption = useMutableCallback((label: string, weight: LivechatPriorityWeight) => {
+		return (
+			<>
+				<PriorityIcon level={weight || LivechatPriorityWeight.NOT_SPECIFIED} showUnprioritized /> {label}
+			</>
+		);
+	});
+
+	return useMemo<ComponentProps<typeof Menu>['options']>(() => {
+		const menuHeading = {
+			type: 'heading' as const,
```

Comment:
```suggestion
			type: 'heading',
```
Does it still require?